posts/innovative-snowflake-features-architecture.md

```diff
@@ -0,0 +1,126 @@
+
+---
+# Snowflake Editions
+Snowflake offers a variety of editions which offer differing levels of service. These include:
+* Standard
+* Premier
+* Enterprise
+* Enterprise for Sensitive data (ESD)
+* Virtual Private Snowflake (VPS)
+
+Look at [Snowflake Editions](https://docs.snowflake.net/manuals/user-guide/intro-editions.html "Snowflake Editions") in the Snowflake Documentation for more information.
+
+---
+# Why Snowflake?
+Snowflake uses a central data repository for persisted data that is accessible to all compute nodes and processes queries in Massively Parallel Processing compute clusters referred to as Virtual Warehouses. This unique architecture allows both for easy data management as well as performance benefits and increased scalability.
+
+In addition, Snowflake is completely ANSI SQL compliant.
+
+---
+# Architecture:
+Snowflake's Architecture consists of the three layers stated below. In each of the following sections, I'll talk a little bit about what each layer contributes to Snowflake.
+
+The Three Layers:
+1. [Database Storage](## 1. Database Storage)
+2. [Query Processing](## 2. Query Processing)
+3. [Global Services](## 3. Global Services)
+
+## 1. Database Storage
+Snowflake automatically converts all data stored into an optimized compressed columnar format (Micro-Partitions) and encrypts it using AES-256 strong encryption.
+
+### Micro-Partitions (FDN [^1])
+When data is loaded into Snowflake, it is automatically divided into **micro-partitions**. Micro-partitions are automatically derived, physical data files which can contain between 50 and 500 MB of uncompressed data. Groups of rows in tables are mapped into individual micro-partitions and are organized in columnar fashion. These micro-partitions are not directly accessible or visible.
+
+![Micro-Partitions](https://docs.snowflake.net/manuals/_images/tables-clustered1.png) [source](https://docs.snowflake.net/manuals/_images/tables-clustered1.png)
+
+In the image above, the table has 24 rows which have been divided into 4 micro-partitions, organized and sorted by column. By dividing the data into micro-partitions, Snowflake can first prune micro-partitions not needed for the query and then prune by column for the remaining micro-partitions. This results in fewer records traversed and much faster response times.
+
+### Clustering
+In addition to storing data as micro-partitions, Snowflake also sorts them naturally according to natural dimensions. This *Clustering* is very important to Snowflake data storage, as it can greatly impact query performance. Clustering occurs upon ingestion. As data is loaded, Snowflake co-locates column data with the same values in the same micro-partition if possible.
```

Comment:
```suggestion
In addition to storing data as micro-partitions, Snowflake also sorts them according to natural dimensions. This *Clustering* is very important to Snowflake data storage, as it can greatly impact query performance. Clustering occurs upon ingestion. As data is loaded, Snowflake co-locates column data with the same values in the same micro-partition if possible.
```